pkg/awsutils/awsutils.go

```diff
@@ -823,7 +823,7 @@ func (cache *EC2InstanceMetadataCache) freeENI(eniName string, maxBackoffDelay t
 	}
 
 	// It does take awhile for EC2 to detach ENI from instance, so we wait 2s before trying the delete.
-	time.Sleep(2 * time.Second)
+	time.Sleep(2 * maxBackoffDelay)
```

Comment:
Isn't 1 minute too long of a wait to detach ENI from instance? I can foresee ENI dangling issues when instances are terminated by ASGs.

Reply:
IMO, we should describe the network interfaces to check if it is available instead of random sleep.

Reply:
> Isn't 1 minute too long of a wait to detach ENI from instance? I can foresee ENI dangling issues when instances are terminated by ASGs.

Yes, @nithu0115 is correct. This should stay as `time.Second`. Nithish, if you want to tackle a less naive approach that uses a call to DescribeNetworkInterfaces, feel free to do so in a different PR. Keep in mind, though, that this cache is all about reducing the number of calls to EC2 to prevent API rate limiting :)

Reply:
@nithu0115 Thanks, good catch. I made the sleep configurable in the tests.

I'm not sure we should call EC2 even more just to check, all requests counts against the TPS limit, so we might as well try to delete. When testing, most waits were between 1.8 to 2.5 seconds delay from detach until the ENI could be deleted. I've never seen it faster, but sometimes up to 10 seconds.

Reply:
> I'm not sure we should call EC2 even more just to check, all requests counts against the TPS limit, so we might as well try to delete.

That's true, but logs would have lot of noise about `ENI failed to delete`